diff_test.go

```diff
@@ -454,6 +455,135 @@ func TestApplyDiffAddfile(t *testing.T) {
 	})
 }
 
+func TestApplyToTree(t *testing.T) {
+	repo := createTestRepo(t)
+	defer cleanupTestRepo(t, repo)
+
+	seedTestRepo(t, repo)
+
+	_, treeA := addAndGetTree(t, repo, "file", "a")
+	_, treeB := addAndGetTree(t, repo, "file", "b")
+	_, treeC := addAndGetTree(t, repo, "file", "c")
```

Comment:
just a small request, can the commits and trees returned by this function be `defer ...Free()`d? I know that `TestApplyDiffAddFile()` doesn't do that either, but we should probably fix that later.

Reply:
@lhchavez done! 👍